### PR TITLE
Add Causilo 1.0.1 probabilistic regression wrapper

### PR DESCRIPTION
**Pull request by dooho00**
We would like to add Causilo 1.0.1 to ScoringBench's univariate regression benchmark.

Causilo is our pretrained tabular model with a scikit-learn interface. The wrapper requests the 999 native quantiles plus probabilities 0.0001 and 0.9999 and uses ScoringBench's shared `quantiles_to_distribution` converter. The Causilo package handles exponential tail extrapolation. Point metrics use the converted distribution's mean, following the shared quantile evaluation path.

This PR adds:
- `CausiloWrapper` with eight estimators, seed 42 and automatic CPU/CUDA selection by default.
- The new versioned registry key `causilo_v1` and dependency `causilo==1.0.1`.
- A Causilo factory in the shared wrapper integration suite.

We installed 1.0.1 from PyPI and ran the native benchmark with fresh model instances: Full completed all 505 folds across 101 datasets, and Lite completed all 202 folds. All 47 quality metrics were finite. The eight shared wrapper integration tests also passed. The Full result ledger and regenerated JSON are provided in the [companion result PR](https://github.com/jonaslandsgesell/ScoringBenchOutput/pull/2); no existing model results were overwritten.

To reproduce Causilo-only Full evaluation after installing the benchmark requirements and `causilo==1.0.1`:

```python
from pathlib import Path
from scoringbench.univariate.datasets import get_DATASETS_CONFIG, validate_datasets
from scoringbench.univariate.runner import run_benchmark
from scoringbench.univariate.utils import set_seed
from scoringbench.univariate.wrappers.causilo import CausiloWrapper

set_seed(42)
run_benchmark(
    validate_datasets(get_DATASETS_CONFIG()),
    {"causilo_v1": CausiloWrapper},
    Path("output"),
    n_folds=5, n_repeats_cv=1, seed=42, sample_size=3000,
)
```

Our run used ScoringBench `cc0f4bb`, checkpoint `nums-ai/causilo` at `94f2bd91`, and eight H100 80 GB workers with disjoint dataset sets and one visible GPU per worker. The snippet above runs the same evaluation sequentially.

Code: https://github.com/nums-ai/causilo (Apache-2.0)
Weights: https://huggingface.co/nums-ai/causilo (Causilo License v1.0)

For the leaderboard, heatmaps and plots, could you please display the model as **Causilo**? We have kept `causilo_v1` as the versioned result identifier for traceability. Please let us know the preferred way to configure a separate display name.

Thank you for maintaining ScoringBench. Please let us know if any changes would make the integration easier to review.

### CODE REVIEW

**Comment by jonaslandsgesell** [comment]
Dear @dooho00 

Thank you for your submission and congratulations on the results!

I replicated the results of this PR independently and will update the leaderboard later today.

Kind regards
Jonas

**Comment by dooho00** [comment]
Thank you, @jonaslandsgesell, for independently verifying our results and arranging the leaderboard update! We’re glad the results are consistent on your setup, and really appreciate your work on ScoringBench and the quick turnaround.
